src/hotspot/share/opto/vectornode.hpp

```diff
@@ -1612,6 +1612,19 @@ class VectorRearrangeNode : public VectorNode {
   Node* vec_shuffle() const { return in(2); }
 };
 
+
+// Selects elements from second and third vector based on the indices held in
+// first vector two input vectors based on the indexes held in first vector.
+class SelectFromTwoVectorNode : public VectorNode {
+public:
+  SelectFromTwoVectorNode(Node* index, Node* src1, Node* src2, const TypeVect* vt)
+  : VectorNode(index, src1, src2, vt) {}
+
+  Node* Ideal(PhaseGVN* phase, bool can_reshape);
+  virtual int Opcode() const;
+};
```

Comment:
`index` -> `indexes` because this is a vector, right? Otherwise I'll assume it is a scalar.
Can you do some pseudo-code, that says how exactly the indices are interpreted? What if they are out of bounds? Does it wrap? Or assume they are in bounds? Undefined behaviour?

Reply:
For me good comments here would be immendely valuable, because it helps with other C2 optimizations.